pkg/pillar/cmd/agent.go

```diff
@@ -0,0 +1,59 @@
+package cmd
+
+import (
+	"flag"
+	"github.com/lf-edge/eve/pkg/pillar/agentlog"
+	"github.com/lf-edge/eve/pkg/pillar/pidfile"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// AgentBase - Interface for all agents to use the AgentBase.
+// Each agent is expected to have a Context that implements
+// this interface.
+type AgentBaseIntf interface {
+	AgentBaseContext() *AgentBaseContext
+	AddAgentSpecificCommandLineFlags() func(ctx interface{})
```

Comment:
Why do you need this function to be returned? You Don't need a return value here.

AddAgentSpecificCommandLineFlags()

Reply:
CommandLine --> Lets stick to CLI

AddAgentSpecificCLIFlags()
ProcessAgentSpecificCLIFlags